commit 13db5607acfc815271f155d89bf422ee211befbd
Author: Shanker Donthineni <shankerd@codeaurora.org>
Date:   Thu Feb 2 20:30:03 2017 -0600

    KVM: arm/arm64: vgic: Stop injecting the MSI occurrence twice
    
    commit 0bdbf3b071986ba80731203683cf623d5c0cacb1 upstream.
    
    The IRQFD framework calls the architecture dependent function
    twice if the corresponding GSI type is edge triggered. For ARM,
    the function kvm_set_msi() is getting called twice whenever the
    IRQFD receives the event signal. The rest of the code path is
    trying to inject the MSI without any validation checks. No need
    to call the function vgic_its_inject_msi() second time to avoid
    an unnecessary overhead in IRQ queue logic. It also avoids the
    possibility of VM seeing the MSI twice.
    
    Simple fix, return -1 if the argument 'level' value is zero.
    
    Reviewed-by: Eric Auger <eric.auger@redhat.com>
    Reviewed-by: Christoffer Dall <cdall@linaro.org>
    Signed-off-by: Shanker Donthineni <shankerd@codeaurora.org>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linux-4.9.47/virt/kvm/arm/vgic/vgic-irqfd.c
===================================================================
--- linux-4.9.47.orig/virt/kvm/arm/vgic/vgic-irqfd.c
+++ linux-4.9.47/virt/kvm/arm/vgic/vgic-irqfd.c
@@ -102,6 +102,9 @@ int kvm_set_msi(struct kvm_kernel_irq_ro
 	if (!level)
 		return -1;
 
+	if (!level)
+		return -1;
+
 	return vgic_its_inject_msi(kvm, &msi);
 }
 
Index: linux-4.9.47/dummy/rpi_1301_13db5607acfc815271f155d89bf422ee211befbd.txt
===================================================================
--- /dev/null
+++ linux-4.9.47/dummy/rpi_1301_13db5607acfc815271f155d89bf422ee211befbd.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
